Form/Type/LegacyLegacySettingType.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Craue\ConfigBundle\Form\Type;
+
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+/**
+ * for Symfony < 2.7
+ *
+ * @author Christian Raue <christian.raue@gmail.com>
+ * @copyright 2011-2015 Christian Raue
+ * @license http://opensource.org/licenses/mit-license.php MIT License
+ */
+class LegacyLegacySettingType extends LegacySettingType {
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function setDefaultOptions(OptionsResolverInterface $resolver) {
```

Comment:
you can just define ``setDefaultOptions(OptionsResolverInterface $resolver)`` in the parent class to add the forward compatibility layer directly. It will not trigger a deprecation notice in this case (it was in pre-release 2.7 versions but this was fixed to make it easy to write 2.3+ compatible code)